packages/nx/src/project-graph/plugins/plugin-pool.ts

```diff
@@ -121,61 +121,42 @@ function createWorkerHandler(
               ? [
                   createNodesPattern,
                   (configFiles, ctx) => {
-                    return new Promise(function (res, rej) {
-                      const tx =
-                        pluginName + ':createNodes:' + performance.now();
+                    const tx = pluginName + ':createNodes:' + performance.now();
+                    return registerPendingPromise(tx, pending, () => {
                       worker.send(
                         createMessage({
                           type: 'createNodes',
                           payload: { configFiles, context: ctx, tx },
                         })
                       );
-                      pending.add(tx);
-                      promiseBank.set(tx, {
-                        promise: this,
-                        resolver: res,
-                        rejecter: rej,
-                      });
                     });
                   },
                 ]
               : undefined,
             createDependencies: result.hasCreateDependencies
               ? (opts, ctx) => {
-                  return new Promise(function (res, rej) {
-                    const tx = pluginName + ':createNodes:' + performance.now();
+                  const tx = pluginName + ':createNodes:' + performance.now();
```

Comment:
Since we're changing things here, we should probably update this name as well. 
```suggestion
                  const tx = pluginName + ':createDependencies:' + performance.now();
```

Reply:
Thanks